costmap_2d/plugins/voxel_layer.cpp

```diff
@@ -313,10 +315,11 @@ void VoxelLayer::raytraceFreespace(const Observation& clearing_observation, doub
     double t = 1.0;
 
     // we can only raytrace to a maximum z height
-    if (wpz > max_obstacle_height_)
+    const double max_z = std::min(max_obstacle_height_, size_z_ * z_resolution_ );
```

Comment:
I think we need to look closer at this specific item, as mentioned in the issue.

Reply:
From my point of view, either we apply this change or we simply remove the parameter __z_voxels__, since this value is supposed to be (max_obstacle_height / z_resolution)